pkg/kubelet/kuberuntime/kuberuntime_container_test.go

```diff
@@ -79,9 +79,12 @@ func TestRemoveContainer(t *testing.T) {
 
 	err = m.removeContainer(containerID)
 	assert.NoError(t, err)
-	// Verify container log is removed
 
-	assert.Equal(t, []string{expectedContainerLogPath, expectedContainerLogPathRotated, expectedContainerLogSymlink}, fakeOS.Removes)
+	// Verify container log is removed.
+	// We could not predict the order of `fakeOS.Removes`, so we use `assert.ElementsMatch` here.
```

Comment:
is the remove order really non-deterministic? that's surprising to me

Reply:
@liggitt while we are cleaning the log files, we called `FakeOS.Glob` https://github.com/kubernetes/kubernetes/blob/14a11060a0775ed609f0810898ebdbe737c59441/pkg/kubelet/logs/container_log_manager.go#L196, which would iterate `FakeOS.Files` https://github.com/kubernetes/kubernetes/blob/14a11060a0775ed609f0810898ebdbe737c59441/pkg/kubelet/container/testing/os.go#L123-L127, and `FakeOS.Files` is a map https://github.com/kubernetes/kubernetes/blob/14a11060a0775ed609f0810898ebdbe737c59441/pkg/kubelet/container/testing/os.go#L36, so the order is non-deterministic after #93333.